### DISCUSSION TITLE
Implement process tickets for the customer interface

### DISCUSSION THREAD

**Discussion by svenoe** [RotherOSS/otobo#1020] [zzz_Project and Idea archive]
Since this project is a little larger, and several parties are involved, I propose to use this discussion as the main way to communicate for planning, and I will keep this post up-to-date with our vision on how things should work and look on each screen. I also created a project for this, to be able to better organize the individual tasks related to this (https://github.com/RotherOSS/otobo/projects/2).

---

**CustomerTicketProcess**
Should in principle look similar to CustomerTicketMessage. The question is, how the additional information fields can be implemented in a pretty way. Also ACLs should be implemented, together with the HideShow-functionality. This can probably done in e second step, though.

---

**CustomerTicketZoom**
We think that the best place for the links to the possible actions would be between the header and the articles, as shown in the red box in the picture below. This is just an example taken from the pr of @dk-ciphron.

![Rastergrafik](https://user-images.githubusercontent.com/48322755/117809971-1cd09100-b25f-11eb-9143-3fd44337786e.png)

An idea here was, to have a possibility to perform simple actions, like "accept" something in a single click, without the detour of CustomerTicketProcess.

---

**CustomerDashboard**
There was the suggestion that a tile should be implemented, which collects and shows all open tasks a customer has to do from all open project tickets. We think that this would be very useful.

---

**Open Questions**
1. Is it necessary to have an explicit "Create new process ticket" for the customer user? A process can easily be initiated by the generic interface out of customer tickets with a certain set of parameters, and we don't see a real benefit of having this separation apparent for the customer user.
2. Would it be useful to integrate the action links into the TicketLists? (E.g. behind the three dots which at the moment have no function.)

**Comment by svenoe** [comment]
We received a pr of dk-ciphron (https://github.com/RotherOSS/otobo/pull/990 - thanks, here, too :), which, for now, I merged into the branch [issue-#1002-customerprocess_01](https://github.com/RotherOSS/otobo/tree/issue-%231002-customerprocess_01) to maybe use it as a foundation for further development.

Our biggest question would be, if it is necessary to have this separation between normal ticket creation and process-ticket creation. Some further improvements of the UI would also be needed, but it is functional, and probably a good starting point. I can get into more detail at another time, for now I would prefer to gather some opinions first.

**Comment by joserribeiro26** [comment]
Using two CustomerTicketMessage and CustomerTicketProcess Interface usually leaves the user confused about opening tickets. If it is centered on a single screen, I think it would be a great idea.
But due to the modularization of the process components, I think that the effort is much greater.

**Comment by acorderor** [comment]
I think since there could be many different processes and from different areas it would be a good idea to be able to see them as a tree, the same as the Service Catalog or the Queues where one can see the available proceses listed and expanded based on a prequalification for example Human Resources -> Process 1, Process 2, etc

**Comment by svenoe** [comment]
> But due to the modularization of the process components, I think that the effort is much greater.

@joserribeiro26 - With the hideshow-functionality the CustomerTicketMessage is already quite modular, we feel, and we already have a package which will be included in the standard, which allows to remove the subject, the main text box, and the attachments from the CustomerTicketMessage. The only real difference I'm aware of, is, that still always an article will be created (and those fields will contain standard values, or a summary of the filled in fields of the CustomerTicketMessage), and you will necessarily have a queue selected. But I'm not so sure, whether there are processes where this is absolutely unwanted (and if, we could also remove the necessity to create an article from CustomerTicketMessage - it seems a bit strange to me, though).

It is probably more effort, though, to prepare all of this via ACLs, I agree.

> I think since there could be many different processes and from different areas it would be a good idea to be able to see them as a tree...

@acorderor - Exactly this would be achieved by letting the customer user select a service, e.g., just in the standard way in the customer ticket message, and then having a generic agent start the process in the background if this specific service was selected, without the customer user having to bother with it.

---

In general it wouldn't be much more effort to support both ways, I guess. Actually in the pr of @dk-ciphron the old "create a new process ticket" is already implemented, anyways. But the question is maybe more, if there should be put effort into expanding this route, like the example of @acorderor, or not.

**Comment by acorderor** [comment]
In fact I have many several implementations where the customer itself requests the service via a process ticket, each of the process has its own set of fields to be completed. For that reason my idea of being able to let them choose a path as a tree to be able to fill the right form and avoid the need to having some one re-routing the ticket or asking for additional information. In fact as @joserribeiro26 it gets confusing to have multiple options, but usually my customers prefer to get ride of the standard option and go through the New Process Ticket option as standard, The key is the expandability and the possibility of form customization it offers. Usually customers are not satisfied with the simplicity of the standard new ticket form.

**Reply by svenoe** [reply]
Acorderor - from your answer I'm not sure whether you are aware of the inclusion of the HideShow functionality to OTOBO. Please have a look at https://otobo.de/en/forums/topic/handy-features/#post-8694 (the thread is German, but the pictures in the example are English) - it is an extremly simplified example, but it shows the principle.

**Comment by acorderor** [comment]
Hello,

In fact I didn't put much attention on the function, I always thought it
was only for dynamic Fields not for other fields, but reading again the
description given, it might work. I will give it a try and see what
happens.

Regards

I will double check

El mar, 25 may 2021 a las 2:09, svenoe ***@***.***>)
escribió:

> Acorderor - from your answer I'm not sure whether you are aware of the
> inclusion of the HideShow functionality to OTOBO. Please have a look at
> https://otobo.de/en/forums/topic/handy-features/#post-8694 (the thread is
> German, but the pictures in the example are English) - it is an extremly
> simplified example, but it shows the principle.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/RotherOSS/otobo/discussions/1020#discussioncomment-779907>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABL52OSFU6TMICYPCPTPAE3TPNLK7ANCNFSM45OY6RSQ>
> .
>


-- 

*Alvaro Cordero RetanaConsultor de Tecnologias*
*Tel: 22585757 ext 123*
*Email:* ***@***.***

**Reply by svenoe** [reply]
It is only for dynamic fields, but they make up a large part of the forms for us, and most other fields can be hidden and filled automatically.

**Comment by acorderor** [comment]
Well, that's not our case.. we have many different processes for much
processes, but I will try some ideas and see if we can achieve our goal
based on the Service Catalog and some ACLs.

Best Regards

El mié, 26 may 2021 a las 2:36, svenoe ***@***.***>)
escribió:

> It is only for dynamic fields, but they make up a large part of the forms
> for us, and most other fields can be hidden and filled automatically.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/RotherOSS/otobo/discussions/1020#discussioncomment-785076>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABL52OTDFETK3WSCGSA4523TPSXKJANCNFSM45OY6RSQ>
> .
>


-- 

*Alvaro Cordero RetanaConsultor de Tecnologias*
*Tel: 22585757 ext 123*
*Email:* ***@***.***

**Comment by svenoe** [comment]
@OlliW opened the issues https://github.com/RotherOSS/otobo/issues/1059 for the implementation of a dashboard tile featuring all open process actions for the customer, and https://github.com/RotherOSS/otobo/issues/1058 for the implementation of optional subject/text/attachments for the CustomerTicketMessage.

For the latter I have a general question: Is it wanted, that process tickets can be opened, which have no article attached? If so, it would be necessary, to also make changes to CustomerTicketZoom which expects at least one article and looks a bit broken, if none exists. (I don't expect the change to be big, but maybe one could also just add a generic "process started" or so, to both title and body, if empty - currently we can auto fill the fields with the service name e.g..)

**Reply by OlliW** [reply]
yes, i think a generic article like "process started" will be reasonable

**Comment by acorderor** [comment]
Hello, I had the idea that CustomerTicketProcess Action was already enabled
en latest versions of OTOBO, although I had installed the latest version
and still don't see the option to become Active once enabled a Customer
Process to be launched.

Also tried to manually call it and still does not seem to be working at the
css level.

I would like to know how is that implementation going?



El mié, 7 jul 2021 a las 3:40, OlliW ***@***.***>) escribió:

> yes, think a generic article is reasonable "process started"
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/RotherOSS/otobo/discussions/1020#discussioncomment-973978>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABL52OVGNOPKU42ZFTNBFULTWQOHXANCNFSM45OY6RSQ>
> .
>


-- 

*Alvaro Cordero RetanaConsultor de Tecnologias*
*Tel: 22585757 ext 123*
*Email:* ***@***.***

**Reply by svenoe** [reply]
Hi acorderor,
our plan is and was to have it included in otobo 10.1 which we will release in Q4 this year.
Best regards, Sven

**Comment by acorderor** [comment]
Hello Svenoe, thanks for your answer, thats great, for some reason I
thought it was going to be included in one of the minor releases, but thats
Ok, I'll keep holding on what is there.

Best Regards

El vie, 20 ago 2021 a las 0:10, svenoe ***@***.***>)
escribió:

> Hi acorderor,
> our plan is and was to have it included in otobo 10.1 which we will
> release in Q4 this year.
> Best regards, Sven
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/RotherOSS/otobo/discussions/1020#discussioncomment-1210122>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABL52OSICRLGXITCEMXFMF3T5XWVRANCNFSM45OY6RSQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 

*Alvaro Cordero RetanaConsultor de Tecnologias*
*Tel: 22585757 ext 123*
*Email:* ***@***.***

**Comment by edusperoni** [comment]
Hey all!

To chime in this conversation, we use exclusively CustomerTicketProcess (CustomerTicketMessage is disabled). For our organization that's easier as we can bundle services together and have different fields based on process (tech support might need room and building numbers, general support doesn't). We have multiple internal (agent only) and external (customer only) processes that streamline the way we work, including limiting the available processes depending on the user's OU.

We're trying out the otobo migration and to my surprise CustomerTicketProcess exists but is broken. While I understand that having only CustomerTicketMessage might be easier, the ease of form creation from CustomerTicketProcess is king for us as our instalation is massive (106 processes, 401 services) and we're expanding the otrs/otobo installation to other OUs for easier accessibility of services.

Is CustomerTicketProcess still planned for 10.1?

**Reply by svenoe** [reply]
Hi Eduardo,

the implementation of the process tickets on the customer side is done for 10.1, at least. With the feedback we got until last autumn, we decided, that it is, for the customer, the nicer way to only have one place to open tickets, which is now the CustomerTicketMessage (CTM). The process management on the customer side is now all part of CustomerTicketZoom for existing tickets. Currently we have no plans to change this, although I can imagine that we will make it easier for the admin to configure the circumstances under which a certain process should be started. One thought (which is not a final decision in any way, just a personal idea, currently), would be, to link processes to services which the customer chooses, as we are expanding on those anyways.

In a way your approach and ours is similar, in completely disabling one of the entry points for the customer, but as a general step, this of course could not be done with the CTM. Unfortunately for you that means, that there won't be a way to migrate the system and keep your structure 1:1. If you want to swap to OTOBO, you will have to prepare the CTM using ACLs and then set up generic agents, which under certain conditions (e.g. certain services) start the processes by setting the process and activity dynamic fields. (This last step will certainly be made easier in the foreseeable time.)

Best regards, Sven

**Reply by edusperoni** [reply]
Hi Steven!

Thank you so much for the quick response!

Unfortunately that migration step is unfeasible for us. As I said before, we have over 100 processes and all of them have different fields, different services (or sometimes a single service that is hidden from the customer), different ticket types (a process can start an incident ticket for a service while another process starts an incident ticket for the same service) and many other data we add with each process. There's no easy way for us to link services to processes as a lot of times the same service can fall into different processes based on the user request.

Even if we did create ACLs and Generic Agents, I imagine we'd have to add more than 1 ACL/process and 1 generic agent/process which at the bare minimum would mean 104 ACL, 104 generic agents + rewrite of all the ACL logic already done by process.

That also adds a lot of complexity, as now we can ask process designers (or the agents) to design the process and we import them 1:1 to OTRS and **sometimes** 1 ACL to show/hide the process based on user OU. If we transitioned to CTM a new process would require 1:1 mapping and then 1 ACL for show/hide (sometimes) + 1 ACL for the 1st process screen to show/hide the fields we need (with no way to make a hidden field with data, IIRC) + 1 generic agent.

Although CTP styling is bugged, it's still functional (it seems). Would PRs rewriting the CTP screen to match the CTM screen with the bare minimum CTP functionality be accepted?

**Reply by svenoe** [reply]
Hi Eduardo,

I fear I have to disappoint you again... I was discussing this at least briefly, now, because in principle we are really happy about an active community and contributions. We would however commit to maintaining a part of OTOBO, which we were quite pleased to get rid of, because, as I wrote, in our eyes the two way approach to letting customers create tickets was not a particularly good design choice. Rather than putting resources into that, we would prefer to enhance the CTM to provide missing features. (E.g. we allowed hiding the whole article section via ACLs now, to match that capability of the old CTP.)

Also - you could of course develop a package which enables this feature again - I assume that you only looked at the CTP of the 10.0 branch. In 10.1 we introduced the process management to the CustomerTicketZoom again, and for that we made some larger scale changes to the former, so that by now it is not only the styling, but missing and changed parts all over the CTP, from perl, over the templates, to the js, and it is not functional at all anymore.

I understand that migrating to using ACLs etc. is a large step, and I thought a bit about possible solutions. One, which I admit has some drawbacks, which would have to be solved, but would likely be quite easy to implement, would be the introduction of a new screen (or presumably directly a tile for the CustomerDashboard), which acts as the old entrance of the CTP, i.e. lets the customer choose a process from a list. If this now would directly create a ticket for the customer and redirect him to the CustomerTicketZoom screen for this ticket, he could just like now in OTRS fill out the data necessary for the activity dialog. (One mini feature which I have on my list anyways, is to directly expand the activity dialog form if there is only one present, so that no additional button click is necessary - if you have a look at the 10.1 branch.) The one real drawback I see, is that an empty ticket is created, even if the customer aborts filling out the first dialog somewhere in the middle. This would have to be dealt with in some way. On the plus side, your complete process designs could be left untouched and would just work as is.

Maybe the latter suggestion could be an option for you, I'm sorry, we are aware that we introduced quite a big change here to OTRS, but in the end, we think, disregarding existing systems, it will be the better choice, and also, as I said, if you look at the changes of the 10.1, we moved quite a bit away from the old way, already...
